docs/guides/python/blender-render.mdx

```diff
@@ -0,0 +1,893 @@
+
+ENV UV_COMPILE_BYTECODE=1 UV_LINK_MODE=copy PYTHONPATH=.
+
+WORKDIR /app
+
+RUN --mount=type=cache,target=/root/.cache/uv \
+  --mount=type=bind,source=uv.lock,target=uv.lock \
+  --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+  uv sync --frozen -v --no-install-project --extra ml --no-dev --no-python-downloads
+
+COPY . /app
+
+RUN --mount=type=cache,target=/root/.cache/uv \
+  uv sync --frozen -v --no-dev --extra ml --no-python-downloads
+
+FROM nvidia/cuda:12.6.2-cudnn-runtime-ubuntu24.04
+
+ENV NVIDIA_DRIVER_CAPABILITIES=all
+ENV NVIDIA_REQUIRE_CUDA="cuda>=8.0"
+
+RUN --mount=type=cache,target=/var/cache/apt/archives \
+  apt-get update && apt-get install -y \
+  software-properties-common \
+  build-essential \
+  libxi6 \
+  libglu1-mesa \
+  libgl1 \
+  libglx-mesa0  \
+  libxxf86vm1 \
+  libxkbcommon0 \
+  libsm6 \
+  libxext6 \
+  libxrender1 \
+  libxrandr2 \
+  libx11-6 \
+  xorg \
+  libxkbcommon0 \
+  ffmpeg \
+  wget \
+  curl \
+  ca-certificates && \
+  add-apt-repository ppa:deadsnakes/ppa && \
+  apt-get install -y python3.11 && \
+  ln -sf /usr/bin/python3.11 /usr/local/bin/python3.11 && \
+  rm -rf /var/lib/apt/lists/*
+
+# Blender variables used for specifying the blender version
+ARG BLENDER_OS="linux-x64"
+ARG BL_VERSION_SHORT="4.2"
+ARG BL_VERSION_FULL="4.2.2"
+ARG BL_DL_ROOT_URL="https://mirrors.ocf.berkeley.edu/blender/release"
+ARG BLENDER_DL_URL=${BL_DL_ROOT_URL}/Blender${BL_VERSION_SHORT}/blender-${BL_VERSION_FULL}-${BLENDER_OS}.tar.xz
+
+WORKDIR /app
+
+# Download and unpack Blender
+ADD $BLENDER_DL_URL blender
+
+ARG HANDLER
+
+ENV HANDLER=${HANDLER}
+ENV PYTHONUNBUFFERED=TRUE
+ENV PYTHONPATH="."
+
+# Copy the application from the builder
+COPY --from=builder --chown=app:app /app /app
+
+# Place executables in the environment at the front of the path
+ENV PATH="/app/.venv/bin:$PATH"
+
+# Run the service using the path to the handler
+ENTRYPOINT python -u $HANDLER
+```
+
+Add two dockerignore files to help optimize the size of the image. These will be `python.dockerignore` and `blender.dockerignore`.
+
+```text
+.mypy_cache/
+.nitric/
+.venv/
+nitric-spec.json
+nitric.yaml
+README.md
+```
+
+We can update the `nitric.yaml` file to allow our services and batch jobs to use the custom docker runtime we set up and point to the services directly.
+
+```yaml title:nitric.yaml
+name: blender-render
+services:
+  - match: services/*.py
+    start: uv run watchmedo auto-restart -p *.py --no-restart-on-command-exit -R python -- -u $SERVICE_PATH
+    runtime: python
+
+batch-services:
+  - match: batches/*.py
+    start: uv run watchmedo auto-restart -p *.py --no-restart-on-command-exit -R python -- -u $SERVICE_PATH
+    runtime: blender
+
+runtimes:
+  blender:
+    dockerfile: blender.dockerfile
+  python:
+    dockerfile: python.dockerfile
+```
+
+We'll also need to add `batch-services` as a preview feature.
+
+```yaml title:nitric.yaml
+# !collapse(1:17) collapsed
+name: blender-render
```

Comment:
indentation

Reply:
would a diff + be useful for these?

Reply:
I think cause its adding to and not changing the lines, the default code block is fine (and they can actually copy from it).